src/npg_porch_cli/api.py

```diff
@@ -17,6 +18,159 @@
 # You should have received a copy of the GNU General Public License along with
 # this program. If not, see <http://www.gnu.org/licenses/>.
 
+import os
+from dataclasses import dataclass, field
 
-def add_two(a, b):
-    return a + b
+import requests
+
+PORCH_CLIENT_ACTIONS = {
+    "list_tasks": "tasks",
+    "list_pipelines": "pipelines",
+    "add_pipeline": "pipelines",
+    "add_task": "tasks",
+    "claim_task": "tasks/claim",
+    "update_task": "tasks",
+}
+
+PORCH_STATUSES = ["PENDING", "CLAIMED", "RUNNING", "DONE", "FAILED", "CANCELLED"]
+
+CLIENT_TIMEOUT = (10, 60)
+
+NPG_PORCH_TOKEN_ENV_VAR = "NPG_PORCH_TOKEN"
+
+
+class AuthException(Exception):
+    pass
+
+
+class InvalidValueException(Exception):
+    pass
+
+
+class ServerErrorException(Exception):
+    pass
+
+
+@dataclass
+class PorchRequest:
+
+    porch_url: str
+    ca_cert: str | None = field(default=None)
+    pipeline_name: str | None = field(default=None)
+    pipeline_url: str | None = field(default=None)
+    pipeline_version: str | None = field(default=None)
+
+    def send(self, action: str, task_input: str | None, task_status: str | None):
+
+        if task_status is not None:
+            task_status = task_status.upper()
+        self._validate_request(
+            action=action, task_input=task_input, task_status=task_status
+        )
+
+        method = "GET"
+        pipeline_data = {
+            "name": self.pipeline_name,
+            "uri": self.pipeline_url,
+            "version": self.pipeline_version,
+        }
+        data = None
+
+        if action == "update_task":
+            method = "PUT"
+            data = {
+                "pipeline": pipeline_data,
+                "task_input": task_input,
+                "status": task_status,
+            }
+        elif action.startswith("list") is False:
+            method = "POST"
+            data = pipeline_data
+            if action == "add_task":
+                data = {"pipeline": pipeline_data, "task_input": task_input}
+
+        request_args = {
+            "headers": self._get_request_headers(action),
+            "timeout": CLIENT_TIMEOUT,
+        }
+        if self.ca_cert is None:
+            request_args["verify"] = data
```

Comment:
verify is supposed to be either `Bool` or a path to a CA bundle. Setting it to `data` seems weird, or relying on side effects of the previous ifs.

Reply:
Sorry, a typo. Fixed

Reply:
and all this is simplified now